ecosystem/web-wallet/src/core/mutations/network.tsx

```diff
@@ -0,0 +1,43 @@
+import { FaucetClient } from 'aptos';
+import { LOCAL_NODE_URL } from 'core/constants';
+import useWalletState from 'core/hooks/useWalletState';
+import { getAccountExists } from 'core/queries/account';
+import { AptosNetwork, getFaucetNetworkFromDevnet } from 'core/utils/network';
+import { useMutation, useQueryClient } from 'react-query';
+
+interface UseSwitchNetworkMutationProps {
+  event: AptosNetwork;
+  testnetIsLive: boolean | undefined;
+}
+
+export const useSwitchNetwork = () => {
+  const queryClient = useQueryClient();
+  const { aptosAccount, updateNetworkState, updateWalletState } = useWalletState();
+
+  const mutation = async ({
+    event,
+    testnetIsLive,
+  }: UseSwitchNetworkMutationProps): Promise<void> => {
+    const newAptosNetwork = event;
+    const newFaucetNetwork = getFaucetNetworkFromDevnet(newAptosNetwork);
+    // switching to local testnet
+    if (newAptosNetwork === LOCAL_NODE_URL && !testnetIsLive) {
+      return;
+    }
+    const accountExists = await getAccountExists({
+      address: aptosAccount?.address().hex(),
+      nodeUrl: newAptosNetwork,
+    });
+    if (!accountExists && aptosAccount) {
+      const faucetClient = new FaucetClient(newAptosNetwork, newFaucetNetwork);
```

Comment:
It is surprising that switching the network will create an account. I think this should be separate from the logic of switching the network.

Reply:
@zacharydenton people will be confused if they switch networks and there's no account available for them to transfer money to or they can't use the faucet there